### PR TITLE
[GEO] Fork Lucene's LatLonShape Classes to local lucene package

### PR DESCRIPTION
**Pull request by nknize**
Lucene 7.6 uses a smaller encoding for LatLonShape. This commit forks the LatLonShape classes to Elasticsearch's local lucene package. These classes will be removed on the release of Lucene 7.6.

### CODE REVIEW

**Comment by nknize** [comment]
@imotov having some issues w/ sql-cli and `LatLonShape` forbiddenAPI signatures. Would you mind having a look?

File: server/src/test/java/org/elasticsearch/index/query/GeoShapeQueryBuilderTests.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ protected GeoShapeQueryBuilder doCreateTestQueryBuilder(boolean indexedShape) {
             if (shapeType == ShapeType.LINESTRING || shapeType == ShapeType.MULTILINESTRING) {
                 builder.relation(randomFrom(ShapeRelation.DISJOINT, ShapeRelation.INTERSECTS));
             } else {
-                // LatLonShape does not support CONTAINS:
+                // XLatLonShape does not support CONTAINS:
```
Comment (jpountz) [inline server/src/test/java/org/elasticsearch/index/query/GeoShapeQueryBuilderTests.java]: let's keep just LatLonShape in comments?

**Review by jpountz — COMMENTED** [review]
This might be more a build issue than a sql-cli issue. It looks like the forbidden APIs in `es-server-signatures.txt` are applied to all modules rather than just the server module like the name suggests, which turns out to be an issue for modules that don't depend on lucene-sandbox where LatLonShape lives. Maybe the best way to proceed in the short term is to have a dedicated signatures file for these classes that live in sandbox and add them only in the Gradle build file for the server module (see eg. how rest-high-level-client specific signatures are added in `client/rest-high-level/build.gradle`)? @atorok Any opinion?

**Comment by imotov** [comment]
@nknize found it. Running the rest of precommit to make sure it didn't break anything else and then will push the changes.

**Review by jpountz — APPROVED** [review]
I'm good with Igor's workaround.

**Review by imotov — APPROVED** [review]
LGTM2

**Comment by atorok** [comment]
@jpountz  The reason we configure the server signatures is that it's possible for projects depending on server to use the APIs forbidden in server via transitive dependencies. We could however be smarter about it and only add those signature if the project actually depends on server. I'll send a PR to make this happen.

**Comment by iverase** [comment]
I run the command `./gradlew assemble` and got the following error:

```
> Task :server:javadoc
/Users/ivera/forks/elasticsearch-fork/server/src/main/java/org/apache/lucene/geo/XRectangle2D.java:41: error: unknown tag: lucene.internal
 * @lucene.internal
   ^
/Users/ivera/forks/elasticsearch-fork/server/src/main/java/org/apache/lucene/geo/XTessellator.java:67: error: unknown tag: lucene.experimental
 * @lucene.experimental
   ^
/Users/ivera/forks/elasticsearch-fork/server/src/main/java/org/apache/lucene/document/XLatLonShape.java:55: error: unknown tag: lucene.experimental
 * @lucene.experimental
   ^
3 errors

> Task :server:javadoc FAILED
```

Removing the Lucene tags remove the problem.

**Pull request merged**